src/heroku/orb.yml

```diff
@@ -1,35 +1,109 @@
+      tag:
         type: string
-        default: "x86"
-        description: The target CPU architecture for the heroku binary. Must be one of “x64”, “x86”, “arm”.
-        # enum ypes are not yet implemented
-        # type: enum
-        # enum: [“x64”, “x86”, “arm”]
+        default: "bionic"
+    docker:
+      - image: buildpack-deps:<< parameters.tag >>
 
+commands:
+  install:
     steps:
       - run:
           name: "Install Heroku CLI, if necessary"
-          # TODO: the assumption of wget, etc. below likely wouldn't work in all cases, especially windows.
           command: |
             if [[ $(command -v heroku) == "" ]]; then
-              wget https://cli-assets.heroku.com/heroku-cli/channels/stable/heroku-cli-{{parameters.os}}-{{parameters.arch}}.tar.gz -O heroku.tar.gz
-              mkdir $HOME/heroku
-              tar -xvzf heroku.tar.gz -C $HOME/heroku --strip-components 1
-              cd $HOME/heroku
-              ./install
+              curl https://cli-assets.heroku.com/install.sh | sh
             else
               echo "Heroku is already installed. No operation was performed."
             fi
+  deploy:
+    parameters:
+      app-name:
+        description:
+          "The name of your Heroku App.
+          For backwards compatibility the literal value
+          `$HEROKU_APP_NAME` is the default, so you can
+          easily use this command by setting an environment
+          variable called HEROKU_APP_NAME"
+        type: string
+        default: $HEROKU_APP_NAME
+      api-key:
+        description:
+          "The API key to use. defaulting to the literal value $HEROKU_API_KEY, so you can
+          populate the environment variable HEROKU_API_KEY and not need to declare a value in code."
+        type: string
+        default: $HEROKU_API_KEY
+      branch:
+        type: string
+        description:
+          Deploy the given branch. The default value is your current branch.
+        default: "$CIRCLE_BRANCH"
+      only-branch:
+        type: string
+        description:
+          "If you specify an only-branch, the deploy will not occur for any other branch. The default value,
+          an empty string, will result in the command running for all branches.
+          This is here mostly because for people moving from CircleCI 1.0 setting up a workflow
+          with branch filters may be more than they want to do, and this is a convenient way to filter
+          out deploys for all but one branch (a typical use would be to pass `master` as the value."
+        default: ""
+    steps:
+      - run:
+          name: Deploy branch to Heroku via git push
+          command: |
+            if [ "<< paramaters.only-branch >>" == "" || "${CIRCLE_BRANCH}" == "<< paramaters.only-branch >>" ]; then
+              git push https://heroku:<< parameters.api-key >>@git.heroku.com/<< parameters.app-name >>.git << parameters.branch >>
+            fi
+jobs:
+  deploy:
+    parameters:
+      app-name:
```

Comment:
Not a change, but an observation - this pattern of repeating params across a job and a command that job uses is going to be fairly common, and creates a lot of redundancy. Might be nice to think of a way to create parameter groups that can be reused.

Reply:
Was thinking the same thing while writing this...